File: Data Structures & Algorithms/string-encode-and-decode/submission-6.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        res  = ""

        for word in strs:
            res += str(len(word)) + "#" + word

        return res
    def decode(self, s: str) -> List[str]:
        res = []


        l = r = 0

        while l < len(s):
            r = l
            while s[r] != '#':
                r += 1
            
            wrdLen = int(s[l:r])

            l = r + 1
            r = l + wrdLen

            res.append(s[l:r])

            l = r
        return res
            

File: Data Structures & Algorithms/anagram-groups/submission-6.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        HashMap<String, ArrayList<String>> res = new HashMap<String, ArrayList<String>>();


        for(String s: strs)
        {
            char[] arr = s.toCharArray();

            Arrays.sort(arr);

            String sortedS = new String(arr);

            res.putIfAbsent(sortedS, new ArrayList<String>());

            res.get(sortedS).add(s);
        }

        return new ArrayList<>(res.values());
        
    }
}
